### PR TITLE
chore: Update dependencies and remove unnecessary and transitive ones.

### PR DESCRIPTION
**Pull request by DaleStan**
I can't easily test on Linux, but this has been working on Windows.

It doesn't seem to make #178 either better or worse.

### CODE REVIEW

File: Yafc/lib/windows/SDL2.dll
Comment (have-fun-was-taken) [inline Yafc/lib/windows/SDL2.dll:1]: Have you updated SDL?  Could you please provide the source from where you took it, so we can compare SHA signatures?

Comment (veger) [reply Yafc/lib/windows/SDL2.dll:1]: I suppose these are the sources? https://github.com/have-fun-was-taken/yafc-ce/issues/178#issuecomment-2187967722

**Comment by veger** [comment]
I tried on Linux,  YAFC runs without errors, I opened my pY projects and all seemed fine.

File: Yafc/lib/windows/SDL2.dll
Comment (DaleStan) [reply Yafc/lib/windows/SDL2.dll:1]: Yes, that's where I got them from.

Comment (have-fun-was-taken) [reply Yafc/lib/windows/SDL2.dll:1]: The hashes match to the x64 version. All good.

**Review by have-fun-was-taken — APPROVED** [review]

**Pull request merged**